projects/angular2-query-builder/src/lib/query-builder/chrono/chrono-date/chrono-date.component.ts

```diff
@@ -0,0 +1,74 @@
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Component, OnInit, Output, EventEmitter, Input, forwardRef } from '@angular/core';
+import * as chrono from 'chrono-node';
+
+@Component({
+  selector: 'chrono-date',
+  template: `
+  <input id="txt" name="txt" #txt="ngModel" type="text" [(ngModel)]="text" (change)="textUpdate($event)" >
+  <input id="cal" name="cal" #cal="ngModel" type="date" [(ngModel)]="date" (change)="calendarUpdate($event)">
+  <span class="chrono-alert" *ngIf="!textValid && txt.touched">Unable to parse text</span>
```

Comment:
Could this be a warning sign? somewhat like https://www.google.com/search?q=warning+sign&client=firefox-b-e&tbm=isch&source=iu&ictx=1&fir=HYgv4NyzrUNCmM%252CZ4zwGVyq64ohtM%252C%252Fm%252F02c428&vet=1&usg=AI4_-kSPOl0FEbOhNcq2o3pWocMEFinsCg&sa=X&ved=2ahUKEwj4yKuqrJ7tAhUlp1kKHUdVBfQQ_B16BAgmEAM&biw=1280&bih=607#imgrc=XbxFzuE138knjM

Reply:
I thought about adding some icons but I decided that it was best to avoid them for now to ensure the UI isn't too cluttered or aggressive.

Reply:
Here's how it looks normally:
![image](https://user-images.githubusercontent.com/73123859/100269769-0d65c180-2f25-11eb-99e6-0764b32db203.png)

Here's how it looks with the parsing error:
![image](https://user-images.githubusercontent.com/73123859/100269813-240c1880-2f25-11eb-9673-e6c39f7b1f10.png)

I followed recommendations from one of the angular forms tutorials I took to not make the warning/error messages too intrusive.

Reply:
Ok, here is what I suggest. The textbox and the datetime of the calendar are redundant. The textbox should occupy all the real estate and only have a button with the calendar. When you use the calendar widget to set a date, it goes into the textbox.

In addition, I would really see a label underneath the textbox that gives you the actual interpretation to make sure it will use the dates you are expecting in the case of relative dates.

So it would be like:

[Today              ]
2020-11-25 00:00:00.000000

Reply:
By default, when using BETWEEN, the timepart is set by default to the beginning of the day (00:00:00...) for the start and end of day (midnight - 1 microsecond) for the end.

Reply:
We should chat about this... hard to explain everything in text.